nixos/modules/services/networking/StevenBlack-hosts.nix

```diff
@@ -0,0 +1,45 @@
+{config, lib, pkgs, ...}:
+with lib;
+{
+  options.services.stevenBlackHosts = {
+    enable = mkEnableOption "filtering hosts via StevenBlack's hosts file";
+    flags = mkOption {
+      description = ''
+        Flags to passs while building the hosts file.
+        Note that compress and skipstatichosts are passed by default.
+        See also <link xlink:href="https://github.com/StevenBlack/hosts#command-line-options>here</link> for details.
+        Note that options that modify directly the hosts file or he global DNS cache will error out
+        as the host file is first built in the standard nix sandbox.
+      '';
+      default = {};
+      #TODO: cli setting type
+      type = with types; attrsOf (
+        oneOf [bool str (listOf bool) (listOf str)]
+      );
+
+    };
+    whitelist = mkOption {
+      description = "List of hosts to allow";
+      default = "";
+      type = types.lines;
```

Comment:
What's the format of this option? This should be documented in the description. Also could there be a more appropriate type for this than just `lines`?

Reply:
I've clarified the description and add an example so that it's clear how to use the option